src/occupational_classification_utils/embed/embedding.py

```diff
@@ -0,0 +1,295 @@
+                #     code = (row["code"])
+                #     docs.append(
+                #         Document(
+                #             page_content=row["text"],
+                #             metadata = {
+                #                 "code": code
+                #             },
+                #         )
+                #     )
+                #     ids.append(str(uuid.uuid3(uuid.NAMESPACE_URL, row["text"])))
+
+        self.vector_store.add_documents(docs, ids=ids)
+        self._index_size = self.vector_store._client.get_collection(  # pylint: disable=protected-access
+            "langchain"
+        ).count()
+        logger.debug(
+            "Inserted %s entries into vector embedding database.", f"{len(docs):,}"
+        )
+
+        # Update shared config
+        embedding_config["index_size"] = self._index_size
+        embedding_config["soc_index"] = soc_index_file or config["lookups"]["soc_index"]
+        embedding_config["soc_structure"] = (
+            soc_structure_file or config["lookups"]["soc_structure"]
+        )
+        embedding_config["soc_condensed"] = (
+            soc_index_file or config["lookups"]["soc_condensed"]
+        )
+        embedding_config["matches"] = self.k_matches
+        embedding_config["db_dir"] = self.db_dir
+        embedding_config["embedding_model_name"] = self.embeddings.model_name
+        embedding_config["llm_model_name"] = config["llm"].get(
+            "llm_model_name", "unknown"
+        )
+
+    def search_index(
+        self, query: str, return_dicts: bool = True
+    ) -> Union[list[dict], list[tuple[Document, float]]]:
+        """Returns k document chunks with the highest relevance to the query.
+
+        Args:
+            query (str): Query string for which the most relevant index entries
+                will be returned.
+            return_dicts (bool, optional): If True, returns data as a list of
+                dictionaries. Otherwise, returns document tuples. Defaults to True.
+
+        Returns:
+            Union[list[dict], list[tuple[Document, float]]]: List of top k index entries
+            by relevance.
+        """
+        top_matches = self.vector_store.similarity_search_with_score(
+            query=query, k=self.k_matches
+        )
+
+        if return_dicts:
+            return [
+                {"distance": float(doc[1])}
+                | {"title": doc[0].page_content}
+                | doc[0].metadata
+                for doc in top_matches
+            ]
+        return top_matches
+
+    def search_index_multi(self, query: list[str]) -> list[dict]:
+        """Returns k document chunks with the highest relevance to a list of query fields.
+
+        Args:
+            query (list[str]): List of query fields (in priority order) for which
+                the most relevant index entries will be returned.
+                Example: [industry_descr, job_title, job_descr].
```

Comment:
Is the query for SOC industry desc, title and job desc or would we just expect it to be job_title?